### PR TITLE
tests p6 and naming for all

### PR DESCRIPTION
**Pull request by Daanvdplas**

### CODE REVIEW

File: src/c2_blockchain/p4_batched_extrinsics.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub struct Header {
     // For example, a hash or a Merkle root.
     extrinsics_root: Hash,
     state: u64,
-    consensus_digest: u64,
+    pub consensus_digest: u64,
```
Comment (JoshOrndorff) [inline src/c2_blockchain/p4_batched_extrinsics.rs:19]: Why does this need to be `pub`? Especially if the other fields on the same struct don't need to be?

File: src/c2_blockchain/p4_batched_extrinsics.rs [hunk 2]
```diff
@@ -152,25 +154,23 @@ fn part_4_child_block() {
...
-fn part_4_child_header() {
+fn bc_4_child_header() {
     let g = Header::genesis();
-    let h1 = g.child(hash(&vec![1, 2, 3]), 6);
+    let h1 = g.child(hash(&[1, 2, 3]), 6);
 
-    assert_eq!(h1.height, 1);
-    assert_eq!(h1.parent, hash(&g));
-    assert_eq!(h1.extrinsics_root, hash(&vec![1, 2, 3]));
+    assert!(g.verify_child(&h1));
```
Comment (JoshOrndorff) [inline src/c2_blockchain/p4_batched_extrinsics.rs:161]: We shouldn't use methods like `verify_child` here because the students will have written those methods.

It may be the case that their code to generate a header is wrong and their code to check a header is also wrong in the same way. So we need to actually check the parent hash and extrinsics root individually.

File: src/c2_blockchain/p5_fork_choice.rs [hunk 3]
```diff
@@ -196,15 +196,39 @@ fn part_5_heaviest_chain() {
...
+            break;
+        }
+    }
+    let mut h_a2 = g.child(2, 0);
+    for i in 0..u64::max_value() {
+        h_a2 = h_a1.child(2, i);
+        if hash(&h_a2) % 2 == 0 {
+            break;
+        }
+    }
```
Comment (JoshOrndorff) [inline src/c2_blockchain/p5_fork_choice.rs:215]: What is the purpose of this change? I'm not fully understanding it. Maybe a comment or two in the test would be helpful?

File: src/c2_blockchain/p6_rich_state.rs [hunk 4]
```diff
@@ -32,7 +33,7 @@ pub struct Header {
     extrinsics_root: Hash,
     /// Stores a cryptographic commitment, like a Merkle root or a hash to the complete
     /// post state.
-    state_root: u64,
+    state_root: Hash,
```
Comment (JoshOrndorff) [inline src/c2_blockchain/p6_rich_state.rs:36]: :100:

**Review by JoshOrndorff — COMMENTED** [review]
Overall looking very good. Just a couple comments to take a look at.

File: src/c2_blockchain/p4_batched_extrinsics.rs (hunk 1 above)
Comment (Daanvdplas) [reply src/c2_blockchain/p4_batched_extrinsics.rs:19]: Because of the mine_extra_hard() in p5 where we have to change the consensus_digest

File: src/c2_blockchain/p4_batched_extrinsics.rs (hunk 2 above)
Comment (Daanvdplas) [reply src/c2_blockchain/p4_batched_extrinsics.rs:161]: Yes of course stupid

File: src/c2_blockchain/p5_fork_choice.rs (hunk 3 above)
Comment (Daanvdplas) [reply src/c2_blockchain/p5_fork_choice.rs:215]: We have to make sure that the hash of the block is even. Otherwise it is random whether this test is correct or not.

Comment (Daanvdplas) [reply src/c2_blockchain/p5_fork_choice.rs:215]: Very simple way of accomplishing though so if you want me to write it cleaner let me know

Comment (JoshOrndorff) [reply src/c2_blockchain/p5_fork_choice.rs:214]: Ah yeah. I get it now. Before the block hashes may have been even or odd. Good change.

**Review by JoshOrndorff — APPROVED** [review]
